python/GangaLHCb/Lib/Applications/BenderBox.py

```diff
@@ -0,0 +1,520 @@
+# 
+#  - For Bender script (the batch analogue of interactive ``bender'' environment):
+#  @code 
+#  j2 = Job()
+#  j2.application = BenderBox ( directory = '$HOME/cmtuser/BenderDev_v30r1' ) 
+#  j2.application.tool = BenderRun ( scripts = [ 'path_to_script/the_script.py' ]  )
+#  @endcode
+#  Note:
+#  1. The ``scripts'' are executed within ``bender'' context
+#  2. ``--no-color'', ``--no-castor'' and ``--batch'' options are added automatically
+#  3. ``data.py'' file is autoimported after other ``import'' scripts 
+#  4. ``summary.xml'' file is *not* added automatically  to the output sandbox
+#
+#  - For Ostap scripts (the batch analogue of interactive ``ostap'' environment):
+#  @code 
+#  j3 = Job()
+#  j3.application = BenderBox ( directory = '$HOME/cmtuser/BenderDev_v30r1' ) 
+#  j3.application.tool = OstapRun ( scripts = [ 'path_to_script/the_script.py' ]  )
+#  @endcode
+#  Note:
+#  1. The ``scripts'' are executed within ``ostap'' context
+#  2. ``--no-color'' and ``--batch'' options are added automatically
+#
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+class BenderBox(GaudiExec):
+    """ Box with Bender-based Applications
+
+    =============
+    Simple Usage:
+    =============
+    j = Job( application = prepareBender ( version = 'v30r1' ) )
+    j.application.tool   = BenderModule  ( module = 'pat_to_module/the_module.py' )
+    
+    ======
+    Usage:
+    ======
+    
+    - For Bender module, the native Bender applictaion:
+    j1 = Job()
+    j1.application = BenderBox ( directory = '$HOME/cmtuser/BenderDev_v30r1' ) 
+    j1.application.tool.module = 'path_to_module/the_module.py' )
+
+    Note:
+    1. The ``module'' is required to have configure&run methods  with the proper signature
+    2. ``summary.xml'' file is not added automatically to the output sandbox
+    
+    - For Bender script (the batch analogue of interactive ``bender'' environment):
+    j2 = Job()
+    j2.application = BenderBox ( directory = '$HOME/cmtuser/BenderDev_v30r1' ) 
+    j2.application.tool = BenderRun ( scripts = [ 'path_to_script/the_script.py' ]  )
+    
+    Note:
+    1. The ``scripts'' are executed within ``bender'' context
+    2. ``--no-color'', ``--no-castor'' and ``--batch'' options are added automatically
+    3. ``data.py'' file is autoimported after other ``import'' scripts 
+    4. ``summary.xml'' file is *not* added automatically  to the output sandbox
+
+     - For Ostap scripts (the batch analogue of interactive ``ostap'' environment):
+    j3 = Job()
+    j3.application = BenderBox( directory = '$HOME/cmtuser/BenderDev_v30r1' ) 
+    j3.application.tool = OstapRun ( scripts = [ 'path_to_script/the_script.py' ]  )
+
+    Note:
+    1. The ``scripts'' are executed within ``ostap'' context
+    2. ``--no-color'' and ``--batch'' options are added automatically
+
+    """
+    _schema = Schema(Version(1, 0), {
```

Comment:
e.g. 
https://github.com/ganga-devs/ganga/blob/develop/python/GangaLHCb/Lib/Backends/Dirac.py#L15

The schema can be inherited from and then expanded as appropriate there is no need to duplicate all that the base class does here.